openfe/setup/molecule.py

```diff
@@ -20,6 +54,8 @@ class Molecule:
         will be used in the hash.
     """
     def __init__(self, rdkit: RDKitMol, name: str = ""):
+        name = _ensure_ofe_name(rdkit, name)
+        _ensure_ofe_version(rdkit)
         self._rdkit = rdkit
         self._hash = hashmol(self._rdkit, name=name)
```

Comment:
I think I'd rather have the ofe_name stuff in the `from_sdf_string` method.

Reply:
My thought is that a `Molecule` should be the same regardless of how it is created, assuming the molecules hash the same. So the following workflows should all produce a molecule that behaves the same:

* User creates RDKit molecule from our SDF output using only RDKit tools, passes that to our `Molecule`
* User loads our SDF output using our helper functions (i.e., `from_sdf_string`)
* User creates RDKit molecule from SMILES string, passes to our `Molecule` with the appropriate `name`

In order to do that, I think we want to ensure that we have the name/version correctly as part of `__init__`, don't we?

Reply:
Ok yeah, I see that `ensure_ofe_name` is required for case 1 and that makes sense